datafusion/core/src/datasource/listing/table.rs

```diff
@@ -893,6 +870,43 @@ impl ListingTable {
             statistics,
         ))
     }
+
+    /// Collects statistics for a given partitioned file.
+    ///
+    /// This method first checks if the statistics for the given file are already cached.
+    /// If they are, it returns the cached statistics.
+    /// If they are not, it infers the statistics from the file and stores them in the cache.
+    async fn do_collect_statistics<'a>(
+        &'a self,
+        ctx: &SessionState,
+        store: &Arc<dyn ObjectStore>,
+        part_file: &PartitionedFile,
+    ) -> Result<Statistics> {
+        let statistics_cache = self.collected_statistics.clone();
+        return match statistics_cache
```

Comment:
I wonder if you need this return here? Maybe it could be something like

```suggestion
        match statistics_cache
```

(you also have to remove the `;` at the end)

I am surprised clippy didn't flag that.